documentation/3_Layers_of_the_Reference_Architecture_Model/3_3_Process_Layer/3_3_5_Publishing_and_using_Data_Apps.md

```diff
@@ -1,8 +1,23 @@
 
-For each Data App that was successfully certified, the corresponding metadata is stored in the App Store for being retrieved by users (e.g., Data Consumers or Data Providers) via a search interface. Searching for a Data App is part of the 'Find App' sub process depicted in Figure **//TODO**. If a user finds a suitable Data App (i.e., matching in functionality and compatible with the user's Connector packaging format) in the App Store, the App can be requested. This is indicated in the 'Retrieve App' sub process, which is conceptually identical with the 'Invoke Data Operation' process outlined in section 3.3.2, which is why a detailed discussion is omitted here.
+After a successful Onboarding Process, App Providers can use their IDS Connector to communicate with an IDS App Store and publish a Data App. Alternatively, providers can publish Data Apps to the App Store without the use of an IDS Connector if one is not available. The "Data App Publication Process" is depicted in Figure [3.3.5.1](#PublishingDataApp) below. In order to be published, certain Data Apps require certification from the Certification Body (see [Section 4.2](../../4_Perspectives_of_the_Reference_Architecture_Model/4_2_Certification_Perspective)), as depicted in the first step of the "Data App Publication Process". For each Data App that was successfully certified, the corresponding metadata and app image are stored in the IDS App Store and can be retrieved by IDS Participants via a search interface offered by the App Store.
+
+![PublishingDataApp](./media/data-app-publication-process.png)
+#### _Fig. 3.3.5.1: "Data App Publication" Process_
+
+When it comes to using a Data App that is offered by an IDS App Store, IDS Participants need to execute the process that is depicted in Figure [3.3.5.2](#UseDataApp). An IDS Participant, here called "App User", can use the search interface of the IDS App Store to look for a suitable Data App through its IDS Connector, as depicted by the "Find Data App" subprocess in Figure [3.3.5.3](#FindDataApp). After the "Find Data App" process is finished, the App User might need to pay for the selected Data App. This is indicated by the "Data App Payment" subprocess, which is conceptually similar to the Contract Negotiation detailed in Section [3.3.3](3_3_3_Contract_Negotiation.md) and can be done directly between the App User and App Provider or through a Clearing House if necessary.
+
+![UseDataApp](./media/use-data-app-process.png)
+#### _Fig. 3.3.5.2: "Use Data App" Process_
+
+![FindDataApp](./media/find-data-app-process.png)
```

Comment:
`App Provider Found` / `No App Provider Found`

Of course, it's good if you can also search for IDS app providers, but first and foremost, don't you want to search/find IDS apps in general?

Reply:
addressed in commit [00996cd](https://github.com/International-Data-Spaces-Association/IDS-RAM_4_0/pull/145/commits/00996cd974b30dd365434dd7bf039a8a9e1257e8)